imod/mf6/boundary_condition.py

```diff
@@ -64,6 +67,15 @@ class BoundaryCondition(Package, abc.ABC):
     not the array input which is used in :class:`Package`.
     """
 
+    def __init__(self, allargs=None):
+        super().__init__(allargs)
+        if allargs["concentration"] is None:
```

Comment:
this is a strange operation. please add a comment as to why it is necessary ( how did "concentration" end up in self.dataset if it was not in the constructor arguments?). Did the wrapper put it in there? But the wrapper used is a property of the derived class, not of boundary_condition.